tests/raster/mosaic/netcdf/test_netcdf.py

```diff
@@ -0,0 +1,78 @@
+import gc
+import os
+from tests.raster.mosaic.mosaic_common import *
+from veranda.raster.mosaic.netcdf import NetCdfReader, NetCdfWriter
+
+
+def test_write_read_image_stack(simple_ds, mosaic, tmp_path):
+    data_var_name = [data_var for data_var in simple_ds.data_vars][0]
+    dst_filepath = os.path.join(tmp_path, "test.nc")
+
+    with NetCdfWriter.from_data(simple_ds, dst_filepath, mosaic=mosaic, stack_dimension='time') as nc_writer:
+        nc_writer.export()
+
+    with NetCdfReader.from_filepaths([dst_filepath]) as nc_reader:
+        assert_reader_data(simple_ds, nc_reader, 0, 0, 10, 10, [data_var_name],
+                           data_variables=[data_var_name])
+        assert_reader_data(simple_ds, nc_reader, 10, 12, 5, 5, [data_var_name],
+                           data_variables=[data_var_name])
+
+    gc.collect()
```

Comment:
As a hint if the test fails, the gc.collect() is not called. You can use a fixture again to handle this
```
@pytest.fixture
def clean_garbage():
    try:
        yield
    finally:
        gc.collect()
```

Even though it returns nothing, you can use it in your test suite, and it will perform the clean-up once the test is finished, similar to a `teardown` call in older test frameworks.

Reply:
Works!